### PR TITLE
DATE_ADD and DATE_SUB function support QUARTER interval

### PR DESCRIPTION
**Pull request by mchades**
## What type of PR is this：
- [x] feature

### CODE REVIEW

**Comment by mchades** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/ArithmeticCommutativeRule.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|

File: gensrc/script/vectorized/vectorized_functions.py [hunk 1]
```diff
@@ -351,6 +351,8 @@
     [50340, 'date_trunc', 'DATETIME', ['VARCHAR', 'DATETIME'], 'TimeFunctions::datetime_trunc', 'TimeFunctions::datetime_trunc_prepare', 'TimeFunctions::datetime_trunc_close'],
     [50350, 'date_trunc', 'DATE', ['VARCHAR', 'DATE'], 'TimeFunctions::date_trunc', 'TimeFunctions::date_trunc_prepare', 'TimeFunctions::date_trunc_close'],
     [50360, 'timestamp', 'DATETIME', ['DATETIME'], 'TimeFunctions::timestamp'],
+    [50370, 'quarters_add', 'DATETIME', ['DATETIME', 'INT'], 'TimeFunctions::quarters_add'],
+    [50371, 'quarters_sub', 'DATETIME', ['DATETIME', 'INT'], 'TimeFunctions::quarters_sub'],
```
Comment (imay) [inline gensrc/script/vectorized/vectorized_functions.py:355]: IMO user can achieve quarters_add by months_add very easily. So better to not introduce these functions.

Comment (mchades) [reply gensrc/script/vectorized/vectorized_functions.py:355]: ok.

**Pull request closed without merging**